nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -961,13 +1103,23 @@ impl DataStore {
     async fn blueprint_current_target_only(
         &self,
         conn: &async_bb8_diesel::Connection<DbConnection>,
+        select_for_update: bool,
```

Comment:
The booleans at the call sites are somewhat confusing. I think it's probably overkill to make this two separate functions, even though that's an option. How about doing what we do elsewhere, and instead of passing `true` or `false` directly, create a local variable named `select_for_update` and pass that in.

Reply:
Yeah I was already borderline on the call site booleans; thanks for the nudge. I don't love the local variable thing, so I replaced this with a two variant enum in 989c5db81 - how does that read?